specification/eventgrid/data-plane/Microsoft.EventGrid/stable/2018-01-01/GeneratedSystemEvents.json

```diff
@@ -2642,6 +2642,71 @@
         }
       }
     },
+    "ApiManagementCircuitBreakerClosedEventData": {
+      "type": "object",
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.CircuitBreaker.Closed event.",
+      "properties": {
+        "backendName": {
+          "type": "string",
+          "description": "Name of the backend for which the circuit has closed."
+        },
+        "circuitBreaker": {
+          "$ref": "#/definitions/ApiManagementCircuitBreakerProperties",
+          "description": "Information related to the circuit breaker configured on the backend."
+        }
+      },
+      "required": [
+        "backendName",
+        "circuitBreaker"
+      ]
+    },
+    "ApiManagementCircuitBreakerOpenedEventData": {
+      "type": "object",
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.ApiManagement.CircuitBreaker.Opened event.",
+      "properties": {
+        "backendName": {
+          "type": "string",
+          "description": "Name of the backend for which the circuit has opened."
+        },
+        "circuitBreaker": {
+          "$ref": "#/definitions/ApiManagementCircuitBreakerProperties",
+          "description": "Information related to the circuit breaker configured on the backend."
+        }
+      },
+      "required": [
+        "backendName",
+        "circuitBreaker"
+      ]
+    },
+    "ApiManagementCircuitBreakerProperties": {
+      "type": "object",
+      "description": "Information related to the circuit breaker configured on the backend.",
+      "properties": {
+        "rules": {
+          "type": "object",
+          "description": "Overview of all configured rules and respective details.",
+          "additionalProperties": {}
```

Comment:
```suggestion
          "additionalProperties": {
            "type": "object"
          }
```

Reply:
Did not realize this was on the generated spec, will fix later on

Reply:
As discussed offline, Typescript looks OK but generation may have a bug. For now, just manually updated the manual Swagger file only

Reply:
Filed https://github.com/Azure/typespec-azure/issues/2554